Code/Framework/AzCore/AzCore/Memory/AllocatorManager.cpp

```diff
@@ -12,96 +12,54 @@
 
 #include <AzCore/Memory/OSAllocator.h>
 #include <AzCore/Memory/AllocationRecords.h>
-#include <AzCore/Memory/MallocSchema.h>
 
 #include <AzCore/std/parallel/lock.h>
 #include <AzCore/std/smart_ptr/make_shared.h>
 #include <AzCore/std/containers/array.h>
 
-namespace AZ::Internal
-{
-    struct AMStringHasher
-    {
-        using is_transparent = void;
-        template<typename ConvertibleToStringView>
-        size_t operator()(const ConvertibleToStringView& key)
-        {
-            return AZStd::hash<AZStd::string_view>{}(key);
-        }
-    };
-    using AMString = AZStd::basic_string<char, AZStd::char_traits<char>, AZStdIAllocator>;
-    using AllocatorNameMap = AZStd::unordered_map<AMString, IAllocator*, AMStringHasher, AZStd::equal_to<>, AZStdIAllocator>;
-    using AllocatorRemappings = AZStd::unordered_map<AMString, AMString, AMStringHasher, AZStd::equal_to<>, AZStdIAllocator>;
-}
-
 namespace AZ
 {
 
-static EnvironmentVariable<AllocatorManager> s_allocManager = nullptr;
-static AllocatorManager* s_allocManagerDebug = nullptr;  // For easier viewing in crash dumps
-
-IAllocator* AllocatorManager::CreateLazyAllocator(size_t size, size_t alignment, IAllocator*(*creationFn)(void*))
+static EnvironmentVariable<AllocatorManager>& GetAllocatorManagerEnvVar()
 {
-    static MallocSchema mallocSchema;
-    void* mem = mallocSchema.Allocate(size, alignment, 0);
-    IAllocator* result = creationFn(mem);
-
-    return result;
+    static EnvironmentVariable<AllocatorManager> s_allocManager;
+    return s_allocManager;
 }
 
+static AllocatorManager* s_allocManagerDebug;  // For easier viewing in crash dumps
+
 //////////////////////////////////////////////////////////////////////////
 bool AllocatorManager::IsReady()
 {
-    return s_allocManager.IsConstructed();
+    return GetAllocatorManagerEnvVar().IsConstructed();
 }
 //////////////////////////////////////////////////////////////////////////
 void AllocatorManager::Destroy()
 {
-    if (s_allocManager)
-    {
-        s_allocManager->InternalDestroy();
-        s_allocManager.Reset();
-    }
 }
 
 //////////////////////////////////////////////////////////////////////////
 // The only allocator manager instance.
 AllocatorManager& AllocatorManager::Instance()
 {
-    if (!s_allocManager)
+    auto& allocatorManager = GetAllocatorManagerEnvVar();
+    if (!allocatorManager)
     {
-        s_allocManager = Environment::CreateVariable<AllocatorManager>(AZ_CRC_CE("AZ::AllocatorManager::s_allocManager"));
+        allocatorManager = Environment::CreateVariable<AllocatorManager>(AZ_CRC_CE("AZ::AllocatorManager::s_allocManager"));
 
-        s_allocManagerDebug = &(*s_allocManager);
+        s_allocManagerDebug = &(*allocatorManager);
     }
 
-    return *s_allocManager;
+    return *allocatorManager;
 }
```

Comment:
It would be nice if this was placed in an AZ::Interface in a future pass on this code.

For the [NameDictionary](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Name/NameDictionary.cpp#L31-L78) we were able to update it to use an AZ::Interface while maintaining backwards compatibility